draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1527,14 +1576,18 @@ as one or more elements of F.
 For the first step, hash\_to\_field calls an auxiliary function expand\_message.
 This document defines two variants of expand\_message: one appropriate
 for hash functions like SHA-2 {{FIPS180-4}} or SHA-3 {{FIPS202}}, and another
-appropriate for extensible-output functions such as SHAKE-128 {{FIPS202}}.
+appropriate for extensible-output functions such as SHAKE128 {{FIPS202}}.
 Security considerations for each expand\_message variant are discussed
 below ({{hashtofield-expand-xmd}}, {{hashtofield-expand-xof}}).
 
-Implementors MUST NOT use rejection sampling to generate a uniformly random element of F.
+Implementors MUST NOT use rejection sampling to generate a uniformly random
+element of F, to ensure that the hash\_to\_field function is amenable to
+constant-time implementation.
 The reason is that rejection sampling procedures are difficult to implement
 in constant time, and later well-meaning "optimizations" may silently render
 an implementation non-constant-time.
+This means that any hash\_to\_field function based on rejection sampling
+would effectively rule out constant-time implementations.
```

Comment:
```suggestion
would prohibit a constant-time implementation.
```

Reply:
It's not *quite* true that it would "prohibit" it---it's possible, just in a painful way. How about "would be incompatible with constant-time implementation"?

Reply:
I'm honestly fine fibbing a little bit here, if only to get the point across. Your proposal also works. (Consider this bike shedding :))